File: leetcode/14/longestCommonPrefix.swift
class Solution {
    func longestCommonPrefix(_ strs: [String]) -> String { 
        let sortedStrs = strs.sorted()
        
        guard let first = sortedStrs.first, let last = sortedStrs.last else {
            return ""
        }

        if first == last{
            return first
        }

        var prefix = first
        while !last.hasPrefix(prefix){
            prefix = String(prefix.dropLast())
        }
        return prefix
    }
}

File: leetcode/14/longestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        answer = ""
        strs.sort()
        first = strs[0]
        last = strs[-1]
        if first == last:
            return first
            
        for i in range(len(first)):
            if first[i] == last[i]:
                answer += first[i]
            else:
                return answer

        return answer
